### PR TITLE
adds FeaturedItem component to feature a selected chart

### PR DESCRIPTION
**Pull request by prydonius**
This adds a component that allows us to feature a specific chart on the
homepage of the Kubeapps Hub.

Currently, the featured chart is hardcoded. It would be trivial to move
this to the override.js configuration that gets pulled down from the
server, allowing us to select the featured chart without code changes.
This work can be done in a follow up PR once this initial implementation
has been worked out.

### CODE REVIEW

File: src/app/featured-item/featured-item.component.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<section class="subhero">
+  <div class="container">
+    <div class="subhero__content">
+      <h2>Featured</h2>
+      <div class="row">
+        <div class="col-2">
+          <img class="subhero__logo" src="/assets/images/jfrog-artifactory-logo.png" height="96" />
```
Comment (andresmgot) [inline src/app/featured-item/featured-item.component.html]: It seems that the icon is a bit misplaced to the right?
image

File: src/app/featured-item/featured-item.component.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+          <p>
+            JFrog Artifactory Open Source is an open source repository manager.
+            JFrog Artifactory was built to tackle some real world problems with
+            Maven, and includes features such as fine-grained permissions, LDAP
+            integration, advanced artifacts management policies, scheduled backups,
+            auditing, and more. JFrog Artifactory works with Maven, Ivy, and Gradle
+            and supports hosting and remote proxying of artifacts. It integrates
+            with Jenkins, TeamCity, Bamboo and other CI servers.
+          </p>
+          <a routerLink="/charts/jfrog/artifactory" class="button button-action">Deploy Now</a>
```
Comment (andresmgot) [inline src/app/featured-item/featured-item.component.html]: I thought this was going to be generated through code (i.e. extracting the description and logo from the chart reference `jfrog/artifactory`) but it's true that we are not sure what we are going to set here so this is fine for the moment.

Comment (prydonius) [reply src/app/featured-item/featured-item.component.html]: That was supposed to be the case for a multiple featured charts, but we needed more content for a single chart. The description we have here is actually pulled from https://bitnami.com/stack/artifactory as https://hub.kubeapps.com/charts/jfrog/artifactory doesn't go into much detail.

What I expect in the future is that we will have a `featured` object in the ConfigMap that will have the structure:

```
featured:
  id: jfrog/artifactory
  description: ...
  logo: ...
```

Once we have that, we could make the use the chart description and logo if description and logo are undefined in the ConfigMap.

File: src/app/featured-item/featured-item.component.html (hunk 1 above)
Comment (prydonius) [reply src/app/featured-item/featured-item.component.html]: The row class this is wrapped it has some padding to the left and right, that's why it's closer to the text. I agree it looks a bit odd, that said @earnubs proposal is WAY better than this, so I'm thinking of merging this as-is and trying to apply his idea on top of the new component we added here in a separate PR.

Comment (andresmgot) [reply src/app/featured-item/featured-item.component.html]: I also like @earnubs proposal so let's implement that.

> so I'm thinking of merging this as-is and trying to apply his idea on top of the new component we added here in a separate PR. What do you think?

I am not sure I am following, you have already added to this PR the style that @earnubs propose, what do you want to do in a different PR?

File: src/app/featured-item/featured-item.component.html (hunk 2 above)
Comment (andresmgot) [reply src/app/featured-item/featured-item.component.html]: Understood, let's add that to the ConfigMap once we have the design validated

File: src/app/featured-item/featured-item.component.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<section class="subhero type-color-reverse">
+  <div class="container">
+    <div class="subhero__content">
+      <h6 class="type-uppercase type-light type-tiny">Featured App</h6>
+      <div class="row margin-reset">
+        <div class="col-6">
+          <h3>JFrog Artifactory</h3>
```
Comment (andresmgot) [inline src/app/featured-item/featured-item.component.html]: It seems that @earnubs is using a different font for this title (at least is more bold). I think the one in the mock looks better, can we change it somehow or this is the only one we can use?

File: src/app/featured-item/featured-item.component.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+          <h6 class="subtitle type-small">Universal Artifact Repository Manager</h6>
+          <p class="type-small">
+            JFrog Artifactory Open Source is an open source repository manager.
+            JFrog Artifactory was built to tackle some real world problems with
+            Maven, and includes features such as fine-grained permissions, LDAP
+            integration, advanced artifacts management policies, scheduled backups,
+            auditing, and more.  
+          </p>
+        </div>
+        <div class="col-6 align-center text-c">
```
Comment (andresmgot) [inline src/app/featured-item/featured-item.component.html]: I would reduce the width of this column a bit, from your screenshot the logo has a lot of margin.

File: src/app/featured-item/featured-item.component.html [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+          <img class="margin-t-giant" src="/assets/images/jfrog-artifactory-logo.png" height="96" />
```
Comment (andresmgot) [inline src/app/featured-item/featured-item.component.html]: white logo:

artifactory-w

File: src/app/featured-item/featured-item.component.html [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        </div>
+        <div class="col-12">
+          <div class="subhero__action margin-t-big margin-b-giant">
+            <a routerLink="/charts/jfrog/artifactory" class="button button-accent button-small">Deploy Now</a>
```
Comment (andresmgot) [inline src/app/featured-item/featured-item.component.html]: would it look better with the gray button?

File: src/app/featured-item/featured-item.component.html (hunk 1 above)
Comment (prydonius) [reply src/app/featured-item/featured-item.component.html]: Sorry, I ended up doing it in this PR and forgot to update the comment!

File: src/app/featured-item/featured-item.component.html (hunk 5 above)
Comment (prydonius) [reply src/app/featured-item/featured-item.component.html]: thank you!!

File: src/app/featured-item/featured-item.component.html (hunk 6 above)
Comment (prydonius) [reply src/app/featured-item/featured-item.component.html]: it does, updated!

**Pull request merged**